python/dgl/data/fraud.py

```diff
@@ -0,0 +1,152 @@
+
+    def __getitem__(self, idx):
+        # get one example by index
+        return self.graphs[idx]
+
+    def __len__(self):
+        # number of data examples
+        return len(self.graphs)
+
+    def save(self):
+        # save processed data to directory `self.save_path`
+        graph_path = os.path.join(self.save_path, self.name + '_dgl_graph.bin')
+        info_path = os.path.join(self.save_path, self.name + '_dgl_graph.pkl')
+        save_graphs(str(graph_path), self.graphs)
+        save_info(info_path, {'label': self.label, 'feature': self.feature})
+
+    def load(self):
+        # load processed data from directory `self.save_path`
+        graph_path = os.path.join(self.save_path, self.name + '_dgl_graph.bin')
+        info_path = os.path.join(self.save_path, self.name + '_dgl_graph.pkl')
+
+        graphs, _ = load_graphs(str(graph_path))
+        info = load_info(str(info_path))
+        self.graphs = graphs
+        self.label = info['label']
+        self.feature = info['feature']
+        self._random_split(self.feature, self.label, self.seed, self.train_size, self.val_size)
+
+    def has_cache(self):
+        # check whether there are processed data in `self.save_path`
+        graph_path = os.path.join(self.save_path, self.name + '_dgl_graph.bin')
+        info_path = os.path.join(self.save_path, self.name + '_dgl_graph.pkl')
+        return os.path.exists(graph_path) and os.path.exists(info_path)
+
+    def _random_split(self, x, node_labels, seed=2, train_size=0.7, val_size=0.1):
+        N = x.shape[0]
+        node_labels = node_labels.transpose(0, 1)
+        if self.name == 'yelp':
+            index = list(range(N))
+            train_idx, test_idx, _, y = train_test_split(index, node_labels, stratify=node_labels, train_size=train_size,
+                                                                    random_state=seed, shuffle=True)
+        elif self.name == 'amazon':
+            # 0-3304 are unlabeled nodes
+            index = list(range(3305, N))
+            train_idx, test_idx, _, y = train_test_split(index, node_labels[3305:], stratify=node_labels[3305:],
+                                                                    test_size=train_size, random_state=seed, shuffle=True)
+        val_idx, test_idx, _, _ = train_test_split(test_idx, y, stratify=y, train_size=val_size/(1-train_size),
+                                                    random_state=seed, shuffle=True)
+        train_mask = torch.zeros(N, dtype=torch.bool)
+        val_mask = torch.zeros(N, dtype=torch.bool)
+        test_mask = torch.zeros(N, dtype=torch.bool)
+        train_mask[train_idx] = True
+        val_mask[val_idx] = True
+        test_mask[test_idx] = True
+        self.train_mask = train_mask
+        self.val_mask = val_mask
+        self.test_mask = test_mask
+    
+class FraudYelpDataset(FraudDataset):
+    def __init__(self, raw_dir=None, random_seed=2, train_size=0.7, val_size=0.1):
```

Comment:
Please refer to the other datasets' description, e.g. https://docs.dgl.ai/_modules/dgl/data/gnn_benckmark.html#AmazonCoBuyComputerDataset. Give some statistics of the data, context of the dataset, and examples of the dataset.

Reply:
Thank you for the review, the statistics, context, and examples of the dataset have been added in latest commit. If there are any further problems, please let me know.